jsettlers.graphics/src/main/java/jsettlers/graphics/map/controls/original/panel/content/settlers/profession/ProfessionPanel.java

```diff
@@ -1,66 +1,245 @@
 package jsettlers.graphics.map.controls.original.panel.content.settlers.profession;
 
-import java.util.function.BiFunction;
+import java.text.MessageFormat;
+import java.util.List;
 
 import go.graphics.text.EFontSize;
+import jsettlers.common.action.Action;
+import jsettlers.common.action.SetMoveableRatioAction;
+import jsettlers.common.images.EImageLinkType;
+import jsettlers.common.images.OriginalImageLink;
 import jsettlers.common.map.IGraphicsGrid;
 import jsettlers.common.map.partition.IPartitionSettings;
+import jsettlers.common.map.partition.IProfessionSettings;
+import jsettlers.common.movable.EMovableType;
+import jsettlers.common.player.IInGamePlayer;
 import jsettlers.common.position.ShortPoint2D;
+import jsettlers.graphics.action.ActionFireable;
+import jsettlers.graphics.localization.Labels;
 import jsettlers.graphics.map.controls.original.panel.content.AbstractContentProvider;
-import jsettlers.graphics.map.controls.original.panel.content.BarFill;
 import jsettlers.graphics.map.controls.original.panel.content.ESecondaryTabType;
 import jsettlers.graphics.map.controls.original.panel.content.updaters.UiContentUpdater;
 import jsettlers.graphics.map.controls.original.panel.content.updaters.UiLocationDependingContentUpdater;
+import jsettlers.graphics.ui.Button;
 import jsettlers.graphics.ui.Label;
+import jsettlers.graphics.ui.Label.EHorizontalAlignment;
+import jsettlers.graphics.ui.UIElement;
 import jsettlers.graphics.ui.UIPanel;
-import jsettlers.graphics.ui.layout.ProfessionsLayout;
 
 public class ProfessionPanel extends AbstractContentProvider implements UiContentUpdater.IUiContentReceiver<IPartitionSettings> {
 
-	private UIPanel panel;
-	private final UiLocationDependingContentUpdater<IPartitionSettings> uiContentUpdater = new UiLocationDependingContentUpdater<>(this::currentDistributionSettingsProvider);
+	private final ContentPanel panel;
+	private final UiLocationDependingContentUpdater<IPartitionSettings> uiContentUpdater;
+	private long lastChangeTimestamp;
 
 	public ProfessionPanel() {
-		panel = new ProfessionsLayout()._root;
-		uiContentUpdater.addListener(this);
+		this.panel = new ContentPanel();
```

Comment:
All the other Panel are defined in .xml files that are "compiled" to java classes by a preprocessor.
I didn't make this system but I would prefer to keep except if there is a good reason why this panel shouldn't be generated by it of course.

Reply:
I have seen the xml way and was very confused. I am really not a fan of it. The ui in settler has not a very complex layout and the additional effort is very high (there is a additional language (xml), there is a additional build step, there is a not type checked relation between xml and java, it is very static). The benefit is only that it calls some constructors for you.

Reply:
Yes, the xml stuff is not ideal and very static but it does the x/y/width/height math for you and it allows easier extension of existing panels.
If we ever come to touch this panel again I might make it a xml file but for now we can leave it as is.